### PR TITLE
Adamedx/x ps exitstatus rebase

### PR DESCRIPTION
**Pull request by adamedx**

### CODE REVIEW

File: spec/functional/resource/powershell_spec.rb [hunk 1]
```diff
@@ -56,14 +56,13 @@
       resource.run_action(:run)
     end
 
-    it "returns the -27 for a powershell script that exits with -27", :windows_powershell_dsc_only do
-      # This is broken on Powershell < 4.0
+    it "returns the exit status 27 for a powershell script that exits with 27" do
       file = Tempfile.new(['foo', '.ps1'])
       begin
-        file.write "exit -27"
+        file.write "exit 27"
```
Comment (jdmundrawala) [inline spec/functional/resource/powershell_spec.rb]: nice

File: lib/chef/provider/powershell_script.rb [hunk 2]
```diff
@@ -24,71 +24,147 @@ class PowershellScript < Chef::Provider::WindowsScript
 
       provides :powershell_script, os: "windows"
 
+      def initialize (new_resource, run_context)
+        super(new_resource, run_context, '.ps1')
+        normalize_script_exit_status
+      end
+
+      def action_run
+        valid_syntax = validate_script_syntax!
```
Comment (jdmundrawala) [inline lib/chef/provider/powershell_script.rb]: couldn't this be `super if validate_script_syntax!`?

**Comment by jdmundrawala** [comment]
:cake:

File: lib/chef/provider/powershell_script.rb (hunk 2 above)
Comment (adamedx) [inline lib/chef/provider/powershell_script.rb]: Now I'm trying to understand why I even need to care about the return value. That's what happens when you leave a PR for a month or two :frowning: .

Seems like I should just let it raise an exception or keep going.

**Comment by adamedx** [comment]
going to close in favor of #3080.

**Pull request closed without merging**